### ISSUE TITLE
DataTable - clickable rows keyboard selection

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7335]
### Expected Behavior
Row in a Datatable that are clickable should be interactive via both 'space' and 'enter'

### Actual Behavior
Trying to interact with the row with 'space' doesn't work, pressing 'enter' sometimes works

### URL, screen shot, or Codepen exhibiting the issue
https://storybook.grommet.io/?path=/story/visualizations-datatable-clickable--clickable-data-table

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by Utsav-Ladani** [comment]
Hi @jcfilben 

Can I work on this issue?

**Comment by jcfilben** [comment]
Hi @Utsav-Ladani feel free to work on this!

**Comment by Utsav-Ladani** [comment]
Hey @jcfilben, I raise a PR #7363 for this bug. Let me know if it requires any futher changes.

**Comment by britt6612** [comment]
@Utsav-Ladani just commented on the PR if we can add a test that would be great! code looks good though

**Comment by Utsav-Ladani** [comment]
Sure @britt6612, I will add unit tests

**Comment by Utsav-Ladani** [comment]
Hi @britt6612,

I added unit tests for `Space` key press. Let me know if it requires any further changes.
